### PR TITLE
LibJS: Add the WeakSet built-in object

### PR DESCRIPTION
**Pull request by IdanHo**
The first two commits fix all WeakSet related test262 test cases. (aside from 5 test cases that rely on other missing js built-ins in libjs)

Note about the third commit: While this (optional as per the spec) "optimization" is pretty important (without it we could end up with infinitely large WeakSets) i feel like my implementation adds too much overhead to the garbage collection pass, so if any heap/garbage collection experts want to chime in with improvements, it would be greatly appreciated :)

### CODE REVIEW

File: Userland/Libraries/LibJS/Tests/builtins/WeakSet/WeakSet.prototype.add.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+test("basic functionality", () => {
+    expect(WeakSet.prototype.add).toHaveLength(1);
+
+    const weak_set = new WeakSet([{ a: 1 }, { a: 2 }, { a: 3 }]);
+    expect(weak_set.add({ a: 4 })).toBe(weak_set);
+    expect(weak_set.add({ a: 1 })).toBe(weak_set);
+});
+
+test("invalid values", () => {
+    const weak_set = new WeakSet();
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/WeakSet/WeakSet.prototype.add.js:10]: Please use pascalCase in all JS code - same below.

**Review by linusg — CHANGES_REQUESTED** [review]
Looks great! Small change in the test files :)

File: Userland/Libraries/LibJS/Tests/builtins/WeakSet/WeakSet.prototype.add.js (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Tests/builtins/WeakSet/WeakSet.prototype.add.js]: fixed!

Comment (IdanHo) [reply Userland/Libraries/LibJS/Tests/builtins/WeakSet/WeakSet.prototype.add.js]: oh whoops, i messed up the commit, just a sec.

**Pull request merged**